### PR TITLE
feat: cairo keccak hints + integration tests

### LINKED ISSUE #465: Builtin: CairoKeccak builtin
Implement cairo keccak builtin hints and fix integration test with keccak.
Rust implementation: https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/hint_processor/builtin_hint_processor/cairo_keccak/keccak_hints.rs

### PR DESCRIPTION
**Pull request by StringNick**
closes #465

### CODE REVIEW

File: src/hint_processor/cairo_keccak_hints.zig [hunk 1]
```diff
@@ -0,0 +1,564 @@
...
+    for (vec) |n| {
+        if (n != null) {
+            switch (n.?) {
+                .felt => |num| try array.append(try num.intoU64()),
+                else => {},
+            }
+        } else {
+            return CairoVMError.ExpectedIntAtRange;
+        }
+    }
```
Comment (tcoratger) [inline src/hint_processor/cairo_keccak_hints.zig:338]:
```suggestion
    for (vec) |maybe_relocatable| {
        if (maybe_relocatable) |n| {
            switch (n) {
                .felt => |num| try array.append(try num.intoU64()),
                else => {},
            }
        } else {
            return CairoVMError.ExpectedIntAtRange;
        }
    }
```

File: src/hint_processor/cairo_keccak_hints.zig [hunk 2]
```diff
@@ -0,0 +1,564 @@
...
+
+    return array;
+}
+
+pub fn u64ArrayToMayberelocatableVec(allocator: std.mem.Allocator, array: []const u64) !std.ArrayList(MaybeRelocatable) {
+    var arr = std.ArrayList(MaybeRelocatable).init(allocator);
```
Comment (tcoratger) [inline src/hint_processor/cairo_keccak_hints.zig:344]:
```suggestion
    var arr = std.ArrayList(MaybeRelocatable).init(allocator);
    errdefer arr.deinit();
```

File: src/hint_processor/cairo_keccak_hints.zig [hunk 3]
```diff
@@ -0,0 +1,564 @@
...
+
+    for (array) |v| try arr.append(MaybeRelocatable.fromInt(u64, v));
```
Comment (tcoratger) [inline src/hint_processor/cairo_keccak_hints.zig:346]: Can't we use appendSlice here?

Comment (StringNick) [reply src/hint_processor/cairo_keccak_hints.zig:346]: fromInt need to call

**Review by lana-shanghai — APPROVED** [review]
lgtm

**Pull request merged**